runtime/runtime/src/state_viewer/mod.rs

```diff
@@ -132,10 +139,12 @@ impl TrieViewer {
                 let code_len = get_code(state_update, account_id, Some(account.code_hash()))?
                     .map(|c| c.code.len() as u64)
                     .unwrap_or_default();
-                if account.storage_usage() > Self::CONTRACT_STATE_SIZE_LIMIT + code_len {
-                    return Err(errors::ViewStateError::AccountStateTooLarge {
-                        requested_account_id: account_id.clone(),
-                    });
+                if let Some(limit) = self.state_size_limit {
+                    if account.storage_usage() > limit + code_len {
```

Comment:
nit:

```suggestion
                    if account.storage_usage() - code_len > limit {
```

now that `limit` is configurable, this could in theory overflow for large limits. Unlikely, but possible. Moving `code_len` to the left is valid if account.storage_usage() guarantees to include `code_len`. If that's not the case, `-` should be replace with saturating_sub.

Reply:
Unlikely hit but still a good security check: people would either choose some far smaller limit like 50K, 50M, or unlimited `None`,